README.md

```diff
@@ -1,17 +1,101 @@
-# dashd-prometheus-exporter
-![dashd prometheus exporter logo](dashd-prometheus-exporter.png)
+# bitcoind-prometheus-exporter
+Bitcoind wallet metrics **Prometheus** exporter.
 
-Dash headless wallet daemon exporter for Prometheus.
+> `bitcoind-prometheus-exporter` is compatible with most **bitcoin** forks.
+
+Produce **wallet** metrics, own **addresses** but also **blockchain** infos.
+Most important metrics are:
+* wallet total balance
+* wallet version
+* if the wallet is unlocked
+* available (spendable) balance for each managed addresses (and watched addresses)
+* best block time and index
+* ...
 
 ## Usage
+Edit the `.env` environment file to suit your needs and run:
+```
+npm start
 ```
-docker run --name my-exporter -p 3000:3000 -e "rpcuser=myrpcuser" -e "rpcpassword=myrpcpassword" -e "rpchost=my-wallet" --link my-wallet lepetitbloc/dashd-prometheus-exporter
+> `bitcoind-prometheus-exporter` uses the `bitcoind` **JSON-RPC** API under the hood and need those credentials:
+> ```
+>rpcuser=test
+>rpcpassword=1cf98b57-5i09-4fa1-9c07-2e28cb2cb47b
+>```
+
+## Usage with other wallets
+The following environment variables are available, that should be enough for *any* **bitcoin** forks:
+> ```
+>symbol=DASH
+>rpcuser=test
+>rpcpassword=1cf98b57-5i09-4fa1-9c07-2e28cb2cb47b
+>rpchost=127.0.0.1
+>rpcport=9999
+>rpcscheme=http
+>```
+
+### Docker
+```
+docker run --name my-exporter -p 3000:3000 -e "rpcuser=myrpcuser" -e "rpcpassword=myrpcpassword" -e "rpchost=my-wallet" --link my-wallet lepetitbloc/bitcoind-prometheus-exporter
 ````
 
 >An easy hack could be to directly use your wallet conf to feed your exporter `env`:
 >```
->docker run --name my-exporter -p 3000:3000 -v /path/to/my/conf:/app/.env --link my-wallet lepetitbloc/dashd-prometheus-exporter
+>docker run --name my-exporter -p 3000:3000 -v /path/to/my/conf:/app/.env --link my-wallet lepetitbloc/bitcoind-prometheus-exporter
 >```
 
+## Example
+When visiting the metrics URL http://localhost:3000/metrics the following **metrics** are produced:
+```
+# HELP best_block_index The block height or index
+# TYPE best_block_index gauge
+best_block_index 69019
+
+# HELP best_block_timestamp_seconds The block time in seconds since epoch (Jan 1 1970 GMT)
+# TYPE best_block_timestamp_seconds gauge
+best_block_timestamp_seconds 1522746083
+
+# HELP chain_difficulty The proof-of-work difficulty as a multiple of the minimum difficulty
+# TYPE chain_difficulty gauge
+chain_difficulty 3511060552899.72
+
+# HELP wallet_version the wallet version
+# TYPE wallet_version gauge
+wallet_version{symbol="BTC"} 71000
+
+# HELP wallet_balance_total the total balance of the wallet
+# TYPE wallet_balance_total gauge
+wallet_balance_total{unconfirmed="2.7345",immature="0",confirmed="42.73453501"} 42.73453501
+
+# HELP wallet_transactions_total the total number of transactions in the wallet
+# TYPE wallet_transactions_total gauge
+wallet_transactions_total 77
```

Comment:
@josqu4red I'm wondering how these wallet metrics will play together when monitoring many wallets... Should the metric include some kind of identifier? Or would that be handle on the collect side?
For addresses balances, it seems pretty straightforward with the address as the label (even so there could be collision with the same address appearing on 2 different forks, but that would be very unlikely though).

Reply:
@josqu4red ping